addons/website/controllers/form.py

```diff
@@ -237,7 +240,9 @@ def extract_data(self, model, values):
     def insert_record(self, request, model, values, custom, meta=None):
         model_name = model.sudo().model
         if model_name == 'mail.mail':
-            email_from = _('"%s form submission" <%s>') % (request.env.company.name, request.env.company.email)
+            if not request.env.company.email:
+                _logger.info('Form submitted by email: company email is empty, using the email_to as email_from')
+            email_from = _('"%s form submission" <%s>') % (request.env.company.name, request.env.company.email or values.get('email_to'))
```

Comment:
@Abridbus @xO-Tx I am not sure: I don't think the `email_to` value was valided with the `website_form_signature` at this point? It's probably less important to validate it being used as email_from rather than as email_to, but would still be better.

Reply:
@qsm-odoo Hmm... I think we can simply move the code of the `mail_to` validation before the `insert_record()` call (probably even before the `extract_data()` to prevent any possibility of inconsistency between `kwargs.get('email_to')` and `data['record'].get('email_to')`). I'll adapt the diff here :+1:
cc @Abridbus